From 53b0300f9e1906f15a550492bad132923e3b33cf Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Mon, 30 Nov 2015 11:46:00 +0800
Subject: [PATCH 0712/5242] MLK-11918-4 video: mxc ipuv3 fb: Fix dangling
 pointer dereference in check_var

commit  d2339642196c64ea6c892434653b56b2be7df43b from
https://source.codeaurora.org/external/imx/linux-imx.git

The function fbi_to_pixfmt() could access the pointer fbi->device provided
by it's input parameter fbi, so let's make sure the pointer isn't dangling.

This patch fixes the following issue reported by Coverity:
Uninitialized pointer read (UNINIT)
uninit_use_in_call: Using uninitialized value tmp_fbi.device when calling
fbi_to_pixfmt.
		switch (fbi_to_pixfmt(&tmp_fbi, true)) {
		case IPU_PIX_FMT_YUV420P2:
		case IPU_PIX_FMT_YVU420P:

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_ipuv3_fb.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
index 6fe8d0f..38ea948 100644
--- a/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/fbdev/mxc/mxc_ipuv3_fb.c
@@ -1647,6 +1647,7 @@ static int mxcfb_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 		fr_h = var->yres_virtual;
 	}
 
+	tmp_fbi.device = info->device;
 	tmp_fbi.var = *var;
 	tmp_fbi.par = mxc_fbi;
 	if (ipu_pixel_format_is_gpu_tile(var->nonstd)) {
-- 
1.7.9.5

